h2o-core/src/main/java/water/rapids/BinaryMerge.java

```diff
@@ -49,15 +50,19 @@
     private final long _base[]; // the col.min() of each column in the key
     private final int _fieldSizes[]; // the widths of each column in the key
     private final int _keySize; // the total width in bytes of the key, sum of field sizes
+    private final BigInteger _baseD[];  // the col.min() of each column in double
```

Comment:
Matt:  

Yes I did.  My original intent was to keep everything in double as long as possible and change the range (row value-base) to long or BigInteger.  However, this does not work because for range with small values, it cannot distinguish those range differences.  Hence back to the drawing board and decide to convert everything to BigInteger right away.